common/app/views/fragments/atoms/storyquestions.scala.html

```diff
@@ -26,6 +26,71 @@
+        <span class="js-storyquestion-atom-id is-hidden" id="@storyquestions.id"></span>
+        <span class="is-hidden" id="js-storyquestion-is-answer-delivery-test-ready" data-is-answer-delivery-test-ready="@isDemandTest"></span>
+
+        @for(questions <- storyquestions.data.editorialQuestions) {
+            @for(qs <- questions) {
+                <h2 id="js-ASK-ACTION-header-@storyquestions.id" class="user__question-title ASK-ACTION-body">
+                    Need something explained?
+                        <span class="user__question-title--secondary">Pick a question: We'll publish an answer to the most popular question shortly</span>
+                </h2>
+                <h2 id="js-SUBMIT-ACTION-header-@storyquestions.id" class="user__question-title SUBMIT-ACTION-body is-hidden">
+                    Thank you for participating
+                    <span class="user__question-title--secondary">
+                    We'll publish an answer to the most popular question shortly. How would you like to get notified when it's ready?
+                    </span>
+                </h2>
+                <h2 id="js-FINAL-ACTION-header-@storyquestions.id" class="user__question-title FINAL-ACTION-body is-hidden">
+                    Thank you for your interest
+                    <span class="user__question-title--secondary">
+                    We're actively developing these notification features but they are not yet built. We're using your feedback to understand how you would like to be notified of the answers. Thanks again.
+                    </span>
+                </h2>
+                <p>
+            <div class="user__question-section"></div>
+                @for(question <- qs.questions) {
+                    <div id="js-ASK-ACTION-body-@question.questionId" class="user__question-container js-ASK-ACTION-body-@storyquestions.id">
+                        <div class="user__questions-text--wrapper js-ask-question-link">
+                            <div class="user__question-text">
+                                <meta class="notranslate" name="js-notranslate-@question.questionId" content="@question.questionText">
+                                <span class="user__questions-text--inner">
+                                @question.questionText
+                                </span>
+                            </div>
+                            <button id="btn-ask-question-@question.questionId" class="user__question-upvote" data-question-id="@question.questionId">
+                                    Ask
+                            </button>
+                        </div>
+                    </div>
+                }
+                <div id="js-SUBMIT-ACTION-body-@storyquestions.id" class="storyquestion-submission-container js-storyquestion-submission-container js-SUBMIT-ACTION-body is-hidden">
+                    <button class="user_answer-delivery btn-answer-delivery-@storyquestions.id" data-delivery-method="email">
```

Comment:
If you do this:
- use a `form` for the `#js-SUBMIT-ACTION-body-@storyquestions.id` elements (and add the `action="#"` attribute)
- for each button, use the `value` attribute instead of `data-delivery-method`, add `type="submit"` and `name="delivery-method"`

Then in your js you can bind to the `submit` event and get the desired value simply this way:
```
form.addEventListener('submit', (evt) => {
  evt.preventDefault();
  const form = evt.target;
  const deliveryMethod = form.elements.namedItem('delivery-method').value
  ...
}
```

Reply:
Thanks! This is helpful and a much nicer way of writing the logic 😅